SupportBank.Console/Program.cs

```diff
@@ -1,11 +1,244 @@
+        {
+            logger.Debug("Parsing JSON file " + filePath);
+            List<Transaction> newTransactions = new List<Transaction>();
+
+            string jsonString = File.ReadAllText(filePath);
+            JArray transactionsJSON = JArray.Parse(jsonString);
+            List<JToken> tokens = transactionsJSON.Children().ToList();
+
+            foreach (var token in tokens)
+            {
+                // read transaction data
+                string date = token["Date"]?.ToString();
+                string fromAccount = token["FromAccount"]?.ToString();
+                string toAccount = token["ToAccount"]?.ToString();
+                string narrative = token["Narrative"]?.ToString();
+                string amount = token["Amount"]?.ToString();
+
+                // build transaction object
+                Transaction transaction = new Transaction(date, fromAccount, toAccount, narrative, amount);
+
+                // validate transaction
+                if (transaction.Amount != 0)
+                    newTransactions.Add(transaction);
+            }
+
+            return newTransactions;
+        }
+
+        private static List<Transaction> ParseXML(string filePath)
+        {
+            logger.Debug("Parsing XML file " + filePath);
+            List<Transaction> newTransactions = new List<Transaction>();
+
+            XmlDocument doc = new XmlDocument();
+            doc.Load(filePath);
+            if (doc.DocumentElement?.ChildNodes == null) return null;
+
+            foreach (XmlNode node in doc.DocumentElement.ChildNodes)
+            {
+                // read transaction data
+                string from = node.SelectSingleNode("Parties/From")?.InnerText;
+                string to = node.SelectSingleNode("Parties/To")?.InnerText;
+                string narrative = node.SelectSingleNode("Description")?.InnerText;
+                string amount = node.SelectSingleNode("Value")?.InnerText;
+
+                double.TryParse(node.Attributes["Date"].InnerText, out var oaDate);
+                string date = DateTime.FromOADate(oaDate).ToString("dd/MM/yyyy");
+
+                // build transaction object
+                Transaction transaction = new Transaction(date, from, to, narrative, amount);
+
+                // validate transaction
+                if (transaction.Amount != 0)
+                    newTransactions.Add(transaction);
+            }
+
+            return newTransactions;
+        }
+
+        // Imports and applies all transactions from file
+        private static void ImportFile(string fileName)
+        {
+            string filePath = "../../../" + fileName;
+
+            // if too short to have a valid extension
+            if (filePath.Length < 4) throw new Exception("Wrong file extension");
+
+            List<Transaction> newTransactions;
+            string fileExtension = filePath.Substring(filePath.Length - 4).ToLower();
+            switch (fileExtension)
```

Comment:
You can use [string.EndsWith](https://docs.microsoft.com/en-us/dotnet/api/system.string.endswith?view=net-6.0#system-string-endswith(system-string)) to avoid the hacky way of checking the extension